tlsutil/config_test.go

```diff
@@ -84,8 +84,8 @@ func TestConfigurator_OutgoingTLS_ServerName(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, tlsConf)
 	require.Len(t, tlsConf.RootCAs.Subjects(), 1)
-	require.Equal(t, tlsConf.ServerName, "consul.example.com")
-	require.False(t, tlsConf.InsecureSkipVerify)
+	require.Empty(t, tlsConf.ServerName)
+	require.True(t, tlsConf.InsecureSkipVerify)
```

Comment:
I was thinking about this test, which I had to change to make it pass.
It is skipping the builtin check from golang tls, but we are still checking the cert chain. Since `VerifyServerHostname` is disabled, we do not check that. I think this is what we want.